static/modules/gamemanager.js

```diff
@@ -243,7 +252,12 @@ class GameManager {
           const collideEntities = map.get(collideTypes[j]);
           if (collideEntities !== undefined) {
             for (let k = 0; k < collideEntities.length; k++) {
-              if (isColliding(targetEntity, collideEntities[k])) {
+              if (
+                !collide(
+                  targetEntity.getCollisionShape(),
+                  collideEntities[k].getCollisionShape()
+                ).isZeroVec()
+              ) {
                 targetEntity.collideWithEntity(collideEntities[k]);
```

Comment:
Does this mean you perform the collision check twice? Once to check if the collide doesn't result in the zero vec, and another time to actually run the collision.

Reply:
No, "Collide" checks to see if it's colliding (and returns the resultant vector) and "collideWithEntity" just triggers the collide event for each object.
However, there is an error here where im not telling the collide method to not resolve the collisions (which speeds it up, less operations), so good catch anyway.